internal/agentdeployer/agent.go

```diff
@@ -163,7 +164,7 @@ func (d *DockerComposeAgentDeployer) SetUp(ctx context.Context, agentInfo AgentI
 	}
 
 	// Service name defined in the docker-compose file
-	agentInfo.Name = dockerTestAgentNamePrefix
+	agentInfo.Name = dockerTestAgentServiceName
```

Comment:
I wonder if with this solution we are still at risk of having the same issue if some package or data stream name is longer.

Could we truncate the package/data stream part?

So for `elastic-agent-carbon_black_cloud-asset_vulnerability_summary-62403` we'd have the truncated version `elastic-agent-carbon_black_cloud-asset_vulnerability_s-62403`.

I would be ok though with applying both changes, the change in the prefix and the change to truncate.

Reply:
Or maybe we could just call it `elastic-agent` plus the run id, that should be unique: `elastic-agent-62403`.

Reply:
The goal to add the package and datastream was to help the user to find the agent in the UI. It's true that as it is right now, there could be issues again if some package or data stream name is longer, so this solution would not work if that happen.

Until now the agent was set as `docker-fleet-agent`. Maybe it's not needed to set in the agent name all that information, since the agents could be found filtering by the Agent Policy. Do you thin that would be good @jsoriano ? 

So I think it could be set just adding as a suffix the RunID: `elastic-agent-12345`

Reply:
Changed here: https://github.com/elastic/elastic-package/pull/1798/commits/a161483576a3c6c42c7eba6947b9ac07e41982fc